library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java

```diff
@@ -318,6 +317,9 @@ public TimeBasedDirectoryScanner()
     @Override
     public void setup(Context.OperatorContext context)
     {
+      if (scanService == null) {
```

Comment:
why is it necessary to check for null? Under what condition it will be not null? As far as I know platform calls setup/terminate once per deployment. IMO, it is safer to create new execution service each time setup is called and possibly assert or throw RuntimeException if scanService is not null to avoid misusage in unit tests.